### PR TITLE
fix: Update tsconfig to address type checking issues

### PR DESCRIPTION
**Pull request by FabricSoul**
**Please describe the changes this PR makes and why it should be merged:**
This PR updates the `tsconfig.json` file by enabling `strictNullChecks`, `noEmit`, and `allowImportingTsExtensions` options. These changes improve the type checking process by ensuring that null and undefined values are handled correctly, preventing type-related runtime errors. Furthermore, the `noEmit` option is set to true to enable type-checking without compiling the code, suitable for our current build process.

**Status and versioning classification:**
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Qjuh** [comment]
How does the `noEmit` suit the build process in `package.json` which only calls `tsc` and nothing else? If it doesn’t emit then there won’t be files to run.

File: packages/create-discord-bot/template/TypeScript/tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
 		"moduleResolution": "NodeNext",
 		"target": "ESNext",
 		"outDir": "dist",
-		"skipLibCheck": true
+		"skipLibCheck": true,
+		"strictNullChecks": true,
+		"allowImportingTsExtensions": true
```
Comment (vladfrangu) [inline packages/create-discord-bot/template/TypeScript/tsconfig.json:13]: This should only be enabled for deno/bun, not every TS project! 👀

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/create-discord-bot/template/TypeScript/tsconfig.json (hunk 1 above)
Comment (FabricSoul) [reply packages/create-discord-bot/template/TypeScript/tsconfig.json:13]: Make sense

File: packages/create-discord-bot/template/TypeScript/tsconfig.json
[the remark this answers is not in the document]
Comment (FabricSoul) [reply packages/create-discord-bot/template/TypeScript/tsconfig.json:1]: do you want me to verify my number?

[the remark this answers is not in the document]
Comment (Qjuh) [reply packages/create-discord-bot/template/TypeScript/tsconfig.json:1]: Please ignore that, they aren’t part of the organization.

**Comment by almeidx** [comment]
Superceded by #10444

**Pull request closed without merging**